groups/bsl/bslstl/bslstl_badoptionalaccess.h

```diff
@@ -0,0 +1,108 @@
+// bslstl_badoptionalaccess.h                                         -*-C++-*-
+#ifndef INCLUDED_BSLSTL_BADOPTIONALACCESS
+#define INCLUDED_BSLSTL_BADOPTIONALACCESS
+
+#include <bsls_ident.h>
+BSLS_IDENT("$Id: $")
+
+//@PURPOSE: Provide an exception class thrown by 'bsl::optional'.
+//
+//@CLASSES:
+//  bsl::bad_optional_access: exception type thrown by 'bsl::optional'
+//
+//@SEE_ALSO: bslstl_optional, bslstl_stdexceptionutil
+//
+//@DESCRIPTION: This component provides a 'bsl::bad_optional_access' exception
+// class.  This exception is thrown by  'bsl::optional::value' when accessing a
+// 'bsl::optional' object that does not contain a value.  If 'std::optional'
+// implementation is available, 'bsl::bad_optional_access' is an alias to
+// 'std::bad_optional_access'.
+//
+
+// Prevent 'bslstl' headers from being included directly in 'BSL_OVERRIDES_STD'
+// mode.  Doing so is unsupported, and is likely to cause compilation errors.
+#if defined(BSL_OVERRIDES_STD) && !defined(BOS_STDHDRS_PROLOGUE_IN_EFFECT)
+#error "include <bsl_memory.h> instead of <bslstl_badoptionalaccess.h> in \
```

Comment:
`bsl_memory.h` seems wrong.

Reply:
hmm... this is pretty much a remnant of me using bslstl_bad_weakptr.h as a source for this file. I don't think this whole #if block. Removed